lib/src/ui/widgets/fancy_dio_tab_view.dart

```diff
@@ -53,22 +53,37 @@ class _FancyDioTabViewState<T extends NetworkBaseModel>
           const Divider(height: 1),
         ],
         Expanded(
-          child: ListView.separated(
-            itemCount: filteredComponents.length,
-            separatorBuilder: (context, index) => const Divider(height: 8),
-            itemBuilder: (context, index) {
-              final filteredComponent = filteredComponents[index];
+          child: widget.components.isEmpty
+              ? Center(
+                  child: Text(
+                    widget.l10nOptions.noDataYetText,
+                    style: Theme.of(context).textTheme.bodyLarge,
+                  ),
+                )
+              : filteredComponents.isEmpty
+                  ? Center(
+                      child: Text(
+                        widget.l10nOptions.noSearchResultsText,
+                        style: Theme.of(context).textTheme.bodyLarge,
+                      ),
+                    )
+                  : ListView.separated(
+                      itemCount: filteredComponents.length,
+                      separatorBuilder: (context, index) =>
+                          const Divider(height: 8),
+                      itemBuilder: (context, index) {
+                        final filteredComponent = filteredComponents[index];
 
-              return Padding(
-                padding: const EdgeInsets.all(16),
-                child: FancyDioTabViewItem(
-                  component: filteredComponent,
-                  l10nOptions: widget.l10nOptions,
-                  tileOptions: widget.tileOptions,
-                ),
-              );
-            },
-          ),
+                        return Padding(
+                          padding: const EdgeInsets.all(16),
+                          child: FancyDioTabViewItem(
+                            component: filteredComponent,
+                            l10nOptions: widget.l10nOptions,
+                            tileOptions: widget.tileOptions,
+                          ),
+                        );
+                      },
+                    ),
```

Comment:
This part seems a bit complicated. Can you make it more simple like exporting widgets?

Reply:
Sure, I hope the new structure makes it a little bit easier to read